pulp_rpm/tests/functional/api/test_copy.py

```diff
@@ -340,6 +342,71 @@ def test_strict_copy_package_to_empty_repo(self):
         for dependency in self.test_package_dependencies:
             self.assertIn(dependency, empty_repo_packages)
 
+    def test_strict_copy_packagecategory_to_empty_repo(self):
+        """Test copy package and its dependencies to empty repository.
+
+        - Create repository and populate it
+        - Create empty repository
+        - Use 'copy' to copy packagecategory recursively
+        - assert packagecategory and its dependencies were copied
+        """
+        empty_repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, empty_repo.pulp_href)
+
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(url=RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        repo = self.repo_api.read(repo.pulp_href)
+        package_category_to_copy = [
+            get_content(repo.to_dict())[PULP_TYPE_PACKAGE_CATEGORY][0]['pulp_href']
+        ]
+        # repository content counts
+        repo_packagecategories_count = len(
+            get_content(repo.to_dict())[PULP_TYPE_PACKAGE_CATEGORY]
+        )
+        repo_packagegroups_count = len(
+            get_content(repo.to_dict())[PULP_TYPE_PACKAGE_GROUP]
+        )
+
+        # do the copy
+        config = [{
+            'source_repo_version': repo.latest_version_href,
+            'dest_repo': empty_repo.pulp_href,
+            'content': package_category_to_copy
+        }]
+        rpm_copy(self.cfg, config, recursive=True)
+        empty_repo = self.repo_api.read(empty_repo.pulp_href)
+
+        # copied repository content counts
+        empty_repo_packages_count = len(
+            get_content(empty_repo.to_dict())[PULP_TYPE_PACKAGE]
+        )
+        empty_repo_packagecategories_count = len(
+            get_content(empty_repo.to_dict())[PULP_TYPE_PACKAGE_CATEGORY]
+        )
+        empty_repo_packagegroups_count = len(
+            get_content(empty_repo.to_dict())[PULP_TYPE_PACKAGE_GROUP]
+        )
+
+        # assert that all dependencies were copy
+        self.assertEqual(
+            repo_packagecategories_count, empty_repo_packagecategories_count
+        )
+        self.assertEqual(
+            repo_packagegroups_count, empty_repo_packagegroups_count
+        )
```

Comment:
It might be good to check that copying a package group only copies the latest package -- by checking the version number.  Is that possible with our fixtures?  If it's not easy, then don't worry about it, but we should document that we need to do it eventually.